lucille-core/src/main/java/com/kmwllc/lucille/connector/CSVConnector.java

```diff
@@ -1,270 +1,56 @@
-    // Either specify the idField, or idFields 
-    if (idField != null) {
-      this.idFields = new ArrayList<String>();
-      this.idFields.add(idField);
-    } else {
-      this.idFields = config.hasPath("idFields") ? config.getStringList("idFields") : new ArrayList<String>();
-    }
-    this.docIdFormat = config.hasPath("docIdFormat") ? config.getString("docIdFormat") : null;
-    // if both a separator char and useTabs is specified, useTabs takes precedence
-    char separator = config.hasPath("separatorChar") ? CharUtils.toChar(config.getString("separatorChar")) : ',';
-    this.separatorChar = (config.hasPath("useTabs") && config.getBoolean("useTabs")) ? '\t' : separator;
-    this.quoteChar = (config.hasPath("interpretQuotes") && !config.getBoolean("interpretQuotes")) ?
-        CSVParser.NULL_CHARACTER : CSVParser.DEFAULT_QUOTE_CHARACTER;
-    this.escapeChar = (config.hasPath("ignoreEscapeChar") && config.getBoolean("ignoreEscapeChar")) ?
-        CSVParser.NULL_CHARACTER : CSVParser.DEFAULT_ESCAPE_CHARACTER;
-    this.lowercaseFields = config.hasPath("lowercaseFields") ? config.getBoolean("lowercaseFields") : false;
-    this.moveToErrorFolder = config.hasPath("moveToErrorFolder") ? config.getString("moveToErrorFolder") : null;
-    this.ignoredTerms = config.hasPath("ignoredTerms") ? config.getStringList("ignoredTerms") : new ArrayList<>();
-    // A directory to move the files to after they are doing being processed.
-    this.moveToAfterProcessing = config.hasPath("moveToAfterProcessing") ? config.getString("moveToAfterProcessing") : null;
+    this.pathStr = config.getString("path");
+    this.csvFileHandler = new CSVFileHandler(config);
   }
 
   @Override
   public void execute(Publisher publisher) throws ConnectorException {
-    if (moveToAfterProcessing != null) {
-      // Create the destination directory if it doesn't exist.
-      File destDir = new File(moveToAfterProcessing);
-      if (!destDir.exists()) {
-        log.info("Creating archive directory {}", destDir.getAbsolutePath());
-        destDir.mkdirs();
-      }
-    }
-
-    if (moveToErrorFolder != null) {
-      File errorDir = new File(moveToErrorFolder);
-      if (!errorDir.exists()) {
-        log.info("Creating error directory {}", errorDir.getAbsolutePath());
-        errorDir.mkdirs();
-      }
-    }
-
-    File pathFile = new File(path);
-    if (!path.startsWith("classpath:") && !pathFile.exists()) {
-      throw new ConnectorException("Path " + path + " does not exist");
-    }
-
-    if (pathFile.isDirectory()) {
-      // no recursion supported
-      for (File f : pathFile.listFiles()) {
-        try {
-          processFile(f.getAbsolutePath(), publisher);
-        } catch (ConnectorException e) {
-          log.warn("Error Processing CSV File. {}", f.getAbsolutePath(), e);
-        }
-      }
-    } else {
-      processFile(path, publisher);
+    File pathFile = new File(pathStr);
```

Comment:
suggest renaming these variables to file, path

I noticed filePath and pathFile in a few other places as well, suggest making this change everywhere

Reply:
resolved in [5d6a758](https://github.com/kmwtechnology/lucille/pull/217/commits/5d6a758f342c3994719490d6a22bb82c110d3545)